.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,19 +1,24 @@
-Add compact, short information about your PR for easier understanding:
+<!-- Add compact, short information about your PR for easier understanding, like the objective of the PR -->
 
-- Goal of the PR
-- How does the PR work?
-- Does it resolve any reported issue?
-- If not a bug fix, why is this PR needed? What usecases does it solve?
+### How does the PR work?
+<!-- Elaborate on the changes, if applicable, or delete this heading -->
 
-## To do
+### Use-cases
+<!-- What use-cases does this PR serve? Delete this heading if this is a bugfix PR -->
```

Comment:
`<!-- Which use cases does this PR serve? Delete this heading if the PR does not implement new features -->`

Reply:
I think `What` would be better suited than `Which` here, as "which" is only used only when there is a finite set of choices. In this context, using "which" would imply that we're presenting a list of use-cases after this sentence, for the PR author to choose from. On the other hand, "what" is more general, and does not imply the existence of limited options to choose from.

Apart from that _trivial_ nit, I've incorporated all your suggestions, incl. those from the comments below.